### PR TITLE
[Java] Catch errors from user callbacks

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes https://github.com/dotnet/aspnetcore/issues/25057

### CODE REVIEW

File: src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java [hunk 1]
```diff
@@ -2565,6 +2565,31 @@ public void sendWithCustomObjectTriggersOnHandlerWithMessagePack()  {
...
+    public void throwFromOnHandlerRunsAllHandlers() {
+        AtomicReference<String> value1 = new AtomicReference<>();
+        AtomicReference<String> value2 = new AtomicReference<>();
+
+        MockTransport mockTransport = new MockTransport();
+        HubConnection hubConnection = TestUtils.createHubConnection("http://example.com", mockTransport);
+
+        hubConnection.on("inc", (param1) -> {
+            value1.set(param1);
+            throw new RuntimeException("throw from on handler");
```
Comment (halter73) [inline src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java:2578]: Is there no way to verify the exception details got logged?

File: src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java [hunk 2]
```diff
@@ -2565,6 +2565,31 @@ public void sendWithCustomObjectTriggersOnHandlerWithMessagePack()  {
...
+        }, String.class);
+        hubConnection.on("inc", (param1) -> {
+            value2.set(param1);
+        }, String.class);
+
+        hubConnection.start().timeout(1, TimeUnit.SECONDS).blockingAwait();
+        mockTransport.receiveMessage("{\"type\":1,\"target\":\"inc\",\"arguments\":[\"Hello World\"]}" + RECORD_SEPARATOR);
+        hubConnection.send("inc", "Hello World");
```
Comment (halter73) [inline src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java:2586]: Why is this necessary? Isn't `mockTransport.receiveMessage(...` what triggers the on callbacks?

File: src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java (hunk 1 above)
Comment (BrennanConroy) [reply src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/HubConnectionTest.java:2578]: I think we have a test logger that I can inspect and assert the logs

File: src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java [hunk 3]
```diff
@@ -223,7 +223,11 @@ Transport getTransport() {
                         List<InvocationHandler> handlers = this.handlers.get(invocationMessage.getTarget());
                         if (handlers != null) {
                             for (InvocationHandler handler : handlers) {
-                                handler.getAction().invoke(invocationMessage.getArguments());
+                                try {
+                                    handler.getAction().invoke(invocationMessage.getArguments());
+                                } catch (Exception e) {
+                                    logger.error("Invoking client side method '{}' failed: {}", invocationMessage.getTarget(), e.getMessage());
```
Comment (halter73) [inline src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java:229]: Would using `e.toString()` (or just `e`) instead allow us to get the exception type and stack trace in the logs similar to C#? If so, I'd prefer that.

```suggestion
                                    logger.error("Invoking client side method '{}' failed: {}", invocationMessage.getTarget(), e);
```

Comment (BrennanConroy) [reply src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java]: > Would using `e.toString()` (or just `e`) instead allow us to get the exception type and stack trace in the logs

Yes, changed it

**Review by halter73 — APPROVED** [review]

**Comment by Pilchie** [comment]
Approved for .NET 5 RC2.
